services/server/src/handlers/topic.ts

```diff
@@ -170,6 +178,46 @@ async function addUserTopics(req: Request, res: Response, next: NextFunction) {
   res.sendStatus(200);
 }
 
+async function replaceUserTopics(req: Request, res: Response, next: NextFunction) {
+  const user = req.user;
+  const allowed = isStudent(user) || isTutor(user) || isTutorManager(user);
+  if (!allowed) return next(forbidden());
+
+  const { removeTopics, addTopics }: ITopic.ReplaceUserTopicsApiPayload = 
+    replaceUserTopicsBodyParser.parse(req.body);
+
+  if (isEmpty(removeTopics) || isEmpty(addTopics)) return next(empty());
+  // to not leave a vulnerability which can make the user exceed the max number
+  if (removeTopics.length !== addTopics.length) return next(bad()); 
+
```

Comment:
- It is fine to one of them to be empty. User might want to just add or remove but not both at the same time.
- No need for the second check (I don't understand it)

```suggestion
  if (isEmpty(removeTopics) && isEmpty(addTopics)) return next(empty());
```

Reply:
I consider just adding or removing topics (by the user) should be handled by other functions (each corresponding handler). using this handler generically is unnecessarily confusing and may be harmful as well. The frontend should use specific and different handlers (endpoints) for different kinds of manipulation: add, remove, update, and replace.

Reply:
Due to how the frontend currently works, this feature is required to work this way. Eventually we will remove the other routes and keep just this new handler.